RockPaperScissors/RockPaperScissors/main.swift

```diff
@@ -1,10 +1,96 @@
 //
 //  RockPaperScissors - main.swift
-//  Created by yagom. 
+//  Created by Lingo, 우롱차.
 //  Copyright © yagom academy. All rights reserved.
 // 
 
 import Foundation
 
-print("Hello, World!")
+enum Settings {
+    static let exitCode = 0
+}
 
+enum GameDisplayMessage {
+    static let menu = "가위(1), 바위(2), 보(3)! <종료 : 0> : "
+    static let gameDidEnd = "게임 종료"
+    static let playerDidWin = "이겼습니다!"
+    static let playerDidLose = "졌습니다!"
+    static let playerDidDraw = "비겼습니다!"
+    static let invalidPlayerInput = "잘못된 입력입니다. 다시 시도해주세요."
+    static let error = "에러가 발생했습니다. 게임을 다시 실행시켜 주세요."
+}
+
+enum HandType: Int, CaseIterable {
+    case scissor = 1
+    case rock = 2
+    case paper = 3
+    
+    static var randomHandType: Int? {
+        return Self.allCases.randomElement()?.rawValue
+    }
+}
+
+enum GameResult {
+    case playerWin
+    case playerLose
+    case playerDraw
+}
+
+func getPlayerInput() -> Int? {
+    guard let playerInput = readLine(),
+          let playerIntInput = Int(playerInput),
+          let playerGameOptionsInput = HandType(rawValue: playerIntInput),
+          (HandType.allCases.contains(playerGameOptionsInput) || playerIntInput == Settings.exitCode) else {
+        return nil
+    }
+    return playerIntInput
+}
+
+func checkGameResult(player playerInput: Int, computer computerInput: Int) -> GameResult {
+    let computerLoseCondition = (computerInput % 3) + 1
+
+    if playerInput == computerLoseCondition {
+        return GameResult.playerWin
+    } else if playerInput == computerInput {
+        return GameResult.playerDraw
+    } else {
+        return GameResult.playerLose
+    }
```

Comment:
오~ 뭔가 새로운 로직으로 코드가 간결해졌군요!
해당 로직이 `스텝2` 에서도 계속 쓰일 수 있을 지는 모르겠지만...
로직 간결화를 위해 계속해서 리팩토링 시도하는 태도가 대단합니다.. 👏🏻 

제안드리고 싶은 건, 해당 메서드는 리턴 타입이 `GameResult`로 명시되어 있잖아요? ㅎㅎ
그 경우, return 해주는 부분에서는 `타입 명시`를 굳이 해주지 않아도 됩니다. ㅎㅎ
쩜(.) 한 번 찍고 사용하면 가독성이 좋아져요~! 😄 

```suggestion
    let computerLoseCondition = (computerInput % 3) + 1

    if playerInput == computerLoseCondition {
        return .playerWin
    } else if playerInput == computerInput {
        return .playerDraw
    } else {
        return .playerLose
    }
```